src/local_scheduler/local_scheduler_client.cc

```diff
@@ -62,13 +90,28 @@ TaskSpec *local_scheduler_get_task(LocalSchedulerConnection *conn,
                                    int64_t *task_size) {
   write_message(conn->conn, MessageType_GetTask, 0, NULL);
   int64_t type;
-  uint8_t *message;
+  uint8_t *task_reply;
   /* Receive a task from the local scheduler. This will block until the local
    * scheduler gives this client a task. */
-  read_message(conn->conn, &type, task_size, &message);
+  read_message(conn->conn, &type, task_size, &task_reply);
+  if (type == DISCONNECT_CLIENT) {
+    LOG_WARN("Exiting because local scheduler closed connection.");
+    exit(1);
+  }
   CHECK(type == MessageType_ExecuteTask);
-  TaskSpec *task = (TaskSpec *) message;
-  return task;
+
+  /* Set the GPU IDs for this task. We only do this for non-actor tasks because
+   * for actors the GPUs are associated with the actor itself and not with the
+   * actor methods. */
+  if (ActorID_equal(conn->actor_id, NIL_ACTOR_ID)) {
```

Comment:
It actually seems like we don't need this if check, but maybe it's better to be explicit.